airtable/emails.js

```diff
@@ -5,165 +5,146 @@
 // To set it up for a city:
 
 // * Copy paste this into the Airtable Scripting Block environment for a city
-// * Update the Global and City-Specific configuration sections below
+// * Update configuration in a table called Email Configuration
 // * Test it out by creating a test delivery and volunteer, and confirming that
 // emails send as expected.
 
+const configTable = base.getTable("Email Configuration");
+const messagesTable = base.getTable("Messages");
+const allConfig = (await base.getTable("Email Configuration").selectRecordsAsync()).records;
+
+const globalConfig = allConfig.filter(c => c.getCellValue("type").name === "global");
+const emailTypesConfig = allConfig.filter(c => c.getCellValue("type").name === "email-template");
+const templateVarsConfig = allConfig.filter(c => c.getCellValue("type").name === "template-variable");
+const allMessages = (await messagesTable.selectRecordsAsync()).records;
+const allVolunteers = (await base.getTable("Volunteers").selectRecordsAsync()).records;
+const allDeliveries = (await base.getTable("Deliveries").selectRecordsAsync()).records;
+
+function globalValueFromKey(key) {
+    return globalConfig.find(k => k.name === key).getCellValue("value");
+}
+
+function findById(table, id) {
+    return table.find(r => r.id === id);
+}
+
 //==========================================================
 // Global configuration -- insert global Nex sendgrid proxy credentials
 //==========================================================
 // This authenticates us to the Nex sendgrid proxy, a small Zeit function
 // that forwards our request to the Sendgrid API.
 // We need to keep this secret because anyone with this credential
 // can send email on our behalf
-const SENDGRID_PROXY_TOKEN= <insert credential here>
+const SENDGRID_PROXY_TOKEN = globalValueFromKey("SENDGRID_PROXY_TOKEN");
 
 //==========================================================
 // City-specific configuration: update to match the city
 //==========================================================
 
 // For each email type, point to the Sendgrid template ID
 // Find the Sendgrid template IDs in Sendgrid UI.
-const SENDGRID_TEMPLATES = {
-    "Matched: to Volunteer": <insert template id>
-    "Matched: to Recipient": <insert template id>
+const EMAIL_TYPES = {};
+for (const emailType of emailTypesConfig) {
+    EMAIL_TYPES[emailType.name] = {
+        "sendgrid_template": emailType.getCellValue("sendgrid_template"),
+        "audience": emailType.getCellValue("audience").name,
+        "send_before_matching": Boolean(emailType.getCellValue("send_before_matching"))
```

Comment:
My prior mental model of "when to send the email" was more like there's various possible stages, of which we're currently just using "upon initial delivery request" and "after matching" ... thoughts on modeling this more that way? Could we do a single-select in the config Base where you pick a "stage"?

Reply:
I was thinking about that too... I want to do something like that, but I think I'd prefer to make that change as a separate change on top of this one, later on. This will work for now, and I'td be good to have a third "stage" use case example in front of me, before implementing a "stages" abstraction, so we have more information about the best way to model it. Does that make sense?

Reply:
Yeah totally makes sense for the code. I suggested some changes to the documentation Google Doc that might make this a bit more clear for people doing config